src/amo/server/base.js

```diff
@@ -254,7 +254,24 @@ function baseServer(
       // banners depending on the user-agent.
       res.vary('User-Agent');
 
-      const history = _createHistory({ req });
+      let history;
+
+      try {
+        history = _createHistory({ req });
+      } catch (error) {
+        // See https://github.com/mozilla/addons-frontend/issues/10061
+        if (error instanceof URIError) {
+          _log.error(`Caught an error during createHistory: ${error}`);
+          return res
+            .status(404)
+            .send(
+              `<!DOCTYPE html>\nWe're sorry, we were unable to parse the request URL: ${error}`,
```

Comment:
mmm, can't we show the normal 404 page?  I'm reading this as it'll show a blank page with literally just this text on it (which doesn't seem any improvement, to the end-user, compared to now)

Reply:
No, we cannot.

Reply:
We discussed this in the issue and decided that this is at least a small improvement over a 500 page or the user, and the fact that a user should never encounter this unless they manually monkey around with a URL. The main purpose of this fix is to eliminate the 500 errors that are causing noise for ops.

Reply:
😿